client/src/main/java/io/pravega/client/segment/impl/SegmentInputStreamFactoryImpl.java

```diff
@@ -20,17 +20,29 @@
 import io.pravega.client.control.impl.Controller;
 import io.pravega.client.security.auth.DelegationTokenProvider;
 import io.pravega.client.security.auth.DelegationTokenProviderFactory;
+import io.pravega.client.stream.Stream;
 import io.pravega.common.MathHelpers;
+import io.pravega.common.util.SimpleCache;
 import io.pravega.shared.security.auth.AccessOperation;
+import java.time.Duration;
+import java.util.Objects;
 import java.util.concurrent.Semaphore;
+import java.util.function.Supplier;
+
 import lombok.RequiredArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 @VisibleForTesting
 @RequiredArgsConstructor
 public class SegmentInputStreamFactoryImpl implements SegmentInputStreamFactory {
-
+    private static final Logger log = LoggerFactory.getLogger(SegmentInputStreamFactoryImpl.class);
+    private static final int CACHE_MAX_SIZE = 100;
+    private static final int EXPIRATION_TIME_MINUTE = 60;
```

Comment:
Is it for every 60 minutes token gets evicted  from cache? and cache is updated with new token in very next call?